lib/pbench/server/api/resources/__init__.py

```diff
@@ -1456,11 +1456,15 @@ def _get_dataset_metadata(
         for i in requested_items:
             if Metadata.is_key_path(i, Metadata.METADATA_KEYS):
                 native_key = Metadata.get_native_key(i)
-                user: Optional[User] = None
+                user_id = None
                 if native_key == Metadata.USER:
                     user = Auth.token_auth.current_user()
+                    if user:
+                        user_id = user.id
```

Comment:
This wouldn't be a bad pattern to encapsulate: it appears several times, and will be completely be replaced, I think, by the new `Auth.token_auth.current_user()` once you change over the Keycloak auth tokens.